hamilflow/models/brownian_motion.py

```diff
@@ -0,0 +1,205 @@
+from functools import cached_property
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import scipy as sp
+from pydantic import BaseModel, computed_field, field_validator
+
+
+class BrownianMotionSystem(BaseModel):
+    r"""Definition of the Brownian Motion system
+
+    For consistency, we always use
+    $\mathbf x$ for displacement, and
+    $t$ for steps. The model we are using is
+
+    $$
+    \begin{align}
+    \mathbf x(t + \mathrm dt) &= \mathrm x(t) +
+    \mathcal{N}(\mu=0, \sigma=\sigma \sqrt{\mathrm d t})
+    \end{align}
+    $$
+
+    References:
+
+    1. Brownian motion and random walks. [cited 13 Mar 2024].
+        Available: https://web.mit.edu/8.334/www/grades/projects/projects17/OscarMickelin/brownian.html
+    2. Contributors to Wikimedia projects. Brownian motion.
+        In: Wikipedia [Internet]. 22 Jan 2024 [cited 13 Mar 2024].
+        Available: https://en.wikipedia.org/wiki/Brownian_motion
+
+    :param sigma: base standard deviation
+        to be used to compute the variance
+    :param delta_t: time granunality of the motion
+    """
+
+    sigma: float
+    delta_t: float = 1
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def gaussian_scale(self) -> float:
+        """The scale (standard deviation) of the Gaussian term
+        in Brownian motion
+        """
+        return self.sigma**2 * self.delta_t
+
+    @field_validator("sigma", "delta_t")
+    @classmethod
+    def check_sigma_delta_t_non_negative(cls, v: float) -> float:
+        if v < 0:
+            raise ValueError(f"Value of sigma and delta_t should be positive: {v=}")
+
+        return v
+
+
+class BrownianMotionIC(BaseModel):
+    """The initial condition for a Brownian motion
+
+    :param x0: initial displacement of the particle,
+        the diminsion of this initial condition determines
+        the dimension of the model too.
+    """
+
+    x0: Union[float, int, List[Union[float, int]]] = 1.0
```

Comment:
Is it necessary to have `Union[float, int]`? What if just `float`?

Reply:
Yeah can be float only.

Reply:
Oh wait, I have this

https://github.com/kausalflow/hamilflow/blob/f06ee378afaec8590f3225605cffb7b5eedc3b1b/hamilflow/models/brownian_motion.py#L70-L71

I think we have to keep the int for consistency.


*Can I remove `int` in this validator?*
I prefer to keep it as int can also be converted to ndarray without any problems. Removing int completely even from this validator will make it rather inconvenient as it throws validator errors when a user uses `1` instead of `1.`